docs/Oscilloscope.md

```diff
@@ -0,0 +1,99 @@
+        <td><img src="/docs/images/oscilloscope_data_analysis.png"></td>
+    </tr>
+</table>
+<table>
+    <tr>
+        <td><img src="/docs/images/oscilloscope_timebase_and_trigger.png"></td>
+        <td><img src="/docs/images/oscilloscope_xy_plot.png"></td>
+    </tr>
+</table>
+
+## Features and Components
+Available features in Pocket Science Lab regarding Oscilloscope are,
+    - 4 channels up to 2MSPS
+    - 2x Sine Wave Generators
+    - 4x PWM generators. 15nS resolution. Up to 8MHz
+
+
+Component | Name in the device
+--- | ---
+Channels | CH1, CH2, CH3, MIC
+Sine waves | SI1, SI2
+PWM Generators | SQ1, SQ2, SQ3, SQ4
+Ground | GND
+
+These features can be used using four components in the Oscilloscope. Note that GND is common for all the pins in the device.
+
+### 1. Channel Parameters
+Channel parameters can be used to select the following settings.
+- Channel/s that needs to be observerd
+- Voltage range to be played in the plot 
+
+When the tick boxes selected from the items below the graph plot, user will be able to find the live plotting of the input read through the relevant channel. 
+For Channel 1 and Channel 2, user can adjust the Voltage range of the graphs 'y axis'. 
+The MIC can be used to read either internal Microphone or External microphone. 
+
+### 2. Timebase and Trigger
+Timebase parameter is used to change the range of 'x axis' which is Time. 
+Trigger takes two parameters channel and voltage. When the read voltage of the selected channel exceeds the selected value for voltage, a trigger occurs.
+
+### 3. Data Analysis
+This function is to fit the read data series into,
+1. Sine function
+2. Square function
+3. Fourier transform
+
+When the oscilloscope reads an input, using data analysis function, user can find out the closest sine/square function for the observed data input.  
+
+### 4. XY Plot
+Using this graph, user can determine the behaviour of one channel input against another channel input. So x and y axes both shows a voltage value for the two channels.  
+
+## How to Use
+
+The input pins for the Oscilloscope are the Channel pins. 
+
+### Observing external waves
+
+Any of the channels must be connected to the positive pin of the signal generator which you want to analyze. Use any of the ground pins in PSLab device to connect the negative/ground pin of the signal generator.
+
+### Observing waves from the in-built Wavegenerator instrument  
+
+Using the Wavegenerator, set values to Sine, Square or Trangular waves. Then the relevant pin you used should be connected to any of the channels. 
+For example, Let's generate two waves, a Sine wave and a Sqaure wave. Set the frequency using Wavegenerator by selecting Sine wave 1 and Square wave 1. Additionally you can change the phase and duty cycle of Square wave. Then connect the Channel 1 and Channel 2 to the Sine wave and Square wave as shown below.
+
+<img src="/docs/images/oscilloscope_schematic.png">
```

Comment:
This image shows a connection of SI pin and SQ pin at the same time. PSLab does not support simultaneous square and sine wave generation